src/app/pages/common/entity/entity-table/entity-table.component.ts

```diff
@@ -418,18 +426,105 @@ export class EntityTableComponent implements OnInit, AfterViewInit, OnDestroy {
       this.isTableEmpty = false;
     } else {
       this.isTableEmpty = true;
-      this.emptyTableConf = {
-        type: EmptyType.no_search_results,
-        large: true,
-        title: T("No Search Results."),
-        message: T(`Your query didn't return any results. Please try again.`)
-      };
+      this.configureEmptyTable(this.dataSource.filter ? EmptyType.no_search_results : this.firstUse ? EmptyType.first_use : EmptyType.no_page_data);
     }
 
     if (this.dataSource.paginator && this.conf.config.paging) {
       this.dataSource.paginator.firstPage();
     }
   }
+
+  configureEmptyTable(emptyType: EmptyType, error: any = null) {
+    if(emptyType && emptyType === EmptyType.loading) {
```

Comment:
This function takes time to understand the logic. Wouldn't it be better to use the "switch" statement on enum for better readability?

Reply:
Good idea. I'll add that in.

Reply:
Thanks!